Upgrade to Tika v1.20

pratheekrebala opened the pull request:
- Updated `tika` to v1.17.
- Updated `node-java` to v0.9.0.
- Increased timeout for FTP test and updated link to test file.
- Updated encrypted docx test to reflect new output from Tika.

All tests are passing.

pratheekrebala committed:
Upgrade to 1.17

Signed-off-by: Pratheek Rebala <pratheekrebala@gmail.com>

chriszs commented:
Just noticed some added warnings:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Jan 19, 2018 3:36:07 AM org.apache.tika.config.InitializableProblemHandler$3 handleInitializableProblem
WARNING: J2KImageReader not loaded. JPEG2000 files will not be processed.
See https://pdfbox.apache.org/2.0/dependencies.html#jai-image-io
for optional dependencies.

Jan 19, 2018 3:36:07 AM org.apache.tika.config.InitializableProblemHandler$3 handleInitializableProblem
WARNING: Tesseract OCR is installed and will be automatically applied to image files unless
you've excluded the TesseractOCRParser from the default parser.
Tesseract may dramatically slow down content extraction (TIKA-2359).
As of Tika 1.15 (and prior versions), Tesseract is automatically called.
In future versions of Tika, users may need to turn the TesseractOCRParser on via TikaConfig.
Jan 19, 2018 3:36:07 AM org.apache.tika.config.InitializableProblemHandler$3 handleInitializableProblem
WARNING: org.xerial's sqlite-jdbc is not loaded.
Please provide the jar on your classpath to parse sqlite files.
See tika-parsers/pom.xml for the correct version.
```

See [this discussion](https://issues.apache.org/jira/browse/TIKA-2389).

Looks like we could either turn these warnings off in tika-config.xml using Tim's patch and/or bundle some of the missing dependencies like J2KImageReader. Node users probably aren't going to want to deal with Java warnings on a package they install, so we may want to make some sane choices for them and turn the warnings off.

Looks like JBIG2 ImageIO was [made optional due to license incompatibility](https://pdfbox.apache.org/2.0/dependencies.html#jai-image-io), but has since been [re-licensed to be compatible](https://github.com/levigo/jbig2-imageio/commit/27dd81cadb4e2bf6759bccca670cfe6b10ac4bd6#diff-409d4384039934c9e806024cd12087ac).

I'm a little unclear on the need for StaticLoggerBinder, does that get used purely for development logging or does it communicate potentially important errors?

We could almost certainly turn the Tesseract warning off in this version, though it will in some future Tika version indicate Tesseract is turned off by default.

I have no idea if bundling SQLite compatibility by default is a good idea or not. The [stated reason](https://tika.apache.org/1.16/api/org/apache/tika/parser/jdbc/SQLite3Parser.html) for making it optional is "potential conflicts of native libraries in web servers."

critocrito committed:
Set a less conspicious user agent and avoid unnecessary 403's.

pratheekrebala committed:
Upgrade Tika to 1.18

Signed-off-by: Pratheek Rebala <pratheekrebala@gmail.com>

pratheekrebala commented:
- Updated tika to v1.18

bnetter committed:
Upgrading to node-java 0.10

pratheekrebala committed:
Merge pull request #1 from bnetter/master

Upgrading to node-java 0.10

pratheekrebala committed:
Merge branch 'browser-user-agent' of git://github.com/critocrito/node-tika into critocrito-browser-user-agent

Signed-off-by: Pratheek Rebala <pratheekrebala@gmail.com>

# Conflicts:
#	jar/node-tika-1.18.jar

pratheekrebala committed:
update tika

Signed-off-by: Pratheek Rebala <pratheekrebala@gmail.com>

pratheekrebala committed:
fix tests

Signed-off-by: Pratheek Rebala <pratheekrebala@gmail.com>

pratheekrebala committed:
Upgrade to Tika 1.20

Also use a local mirror of Apache, apache.org instead of eu.apache.org to download the jar file.

Signed-off-by: Pratheek Rebala <pratheekrebala@gmail.com>

pratheekrebala commented:
- Updated tika to v1.20 in Publici/node-tika@8cd0b05

pratheekrebala commented:
@mattcg Would it be possible to merge this PR? The test that fails is a timeout when Travis is trying to fetch the FTP file. It seems to be unrelated to this package. 

Thank you! 🙏 🙏 🙏
